### DISCUSSION TITLE
Wrong UA_EVENT on incoming call with early video

### DISCUSSION THREAD

**Discussion by frogblueDFi** [baresip/baresip#2887] [Q&A]
baresip/re version: v3.8.1 (and previous at least since v3.5)
Guess, logs, config or network traffic not necessary as issue obvious due to code change.

On an incoming call with early video I do NOT receive a UA_EVENT_CALL_REMOTE_SDP ua event signalling that the remote sends video. But rather a UA_EVENT_CALL_LOCAL_SDP is emitted.

The obvious problem (for me) is a change in file call.c function call_progress_dir, line 1296.
Previous code (baresip v3.2.0)
`
if (call->got_offer) err = update_media(call);
`
was changed to
`
if (call->got_offer) { ua_event(call->ua, UA_EVENT_CALL_LOCAL_SDP, call, "answer"); err = call_update_media(call); }
`

So previous (functional code) sends the UA_EVENT_CALL_REMOTE_SDP via update_media() not UA_EVENT_CALL_LOCAL_SDP.
Reverting the change did the fix for me as I rely on the event to indicate if the remote sends video or not.

Regards
Dietmar

**Comment by alfredh** [comment]
hi Dietmar,

Could you please make a log where the relevant events are printed ?

Also, is it possible to convert that code snippet into a PR ?


Alfred

**Comment by cspiel1** [comment]
`call_progress_dir()` is called locally and initiates a 183 Session Progress to be sent. In my opinion the local application should be informed that we sent an SDP answer. That's why `UA_EVENT_CALL_LOCAL_SDP` with param `answer` is emitted.

The `UA_EVENT_CALL_REMOTE_SDP` with `offer` is emitted when the incoming INVITE is processed.

In order to notice that there is an incoming video your application could wait for the `CALL_RTPESTAB (video)` which signals the first incoming video packet.

Our it could process the `CALL_LOCAL_SDP` which contains many details. E.g. the `videodir`. Here is like module `ctrl_dbus` converts this into a DBUS signal.
```
signal time=1705993632.162117 sender=:1.2384 -> destination=(null destination) serial=5 path=/baresip; interface=com.github.Baresip; member=event
   string "call"
   string "CALL_LOCAL_SDP"
   string "{"type":"CALL_LOCAL_SDP","class":"call","accountaor":"sip:A2@localhost","direction":"incoming","peeruri":"sip:B@kitchen","peerdisplayname":"B","id":"308020807cd78143","remoteaudiodir":"sendrecv","remotevideodir":"sendrecv","audiodir":"inactive","videodir":"recvonly","localaudiodir":"inactive","localvideodir":"recvonly","param":"answer"}"
```

To sum it up:
- `CALL_LOCAL_SDP` informs that we sent SDP to the remote
- `CALL_REMOTE_SDP` informs that we received an SDP from the remote

**Comment by cspiel1** [comment]
In my opinion this is  not an issue. I convert it to a discussion.

**Comment by frogblueDFi** [comment]
Hi, I am little bit confused!

As documentation is rare I assumed:
UA_EVENT_CALL_REMOTE_SDP will signal me if remote will send (or offers) video or not.
UA_EVENT_CALL_LOCAL_SDP will signal if the local site offers video or not.
I checked this with:
`sdp_dir dir = sdp_media_rdir(stream_sdpmedia(video_strm(call_video(call))));`
and
`sdp_dir dir = sdp_media_ldir(stream_sdpmedia(video_strm(call_video(call))));`
This worked perfect with previous versions of baresip (v3.2.0)
As for now the remote offer will not lead to a UA_EVENT_CALL_REMOTE_SDP event but rather a UA_EVENT_CALL_LOCAL_SDP event. Maybe the same check here (sdp_media_rdir ) will also work.
Waiting for UA_EVENT_CALL_RTPESTAB to check if remote offers video sounds not ok for me.
Video transfer could fail for any other reason, I want to know if the remote will try(!) to send video.

Dietmar
